### PR TITLE
Adding Azure Firewall sandbox template with Linux VMs and multiple public IP address support

### PR DESCRIPTION
**Pull request by iatodoro**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

* Adding new template for creating Azure Firewall sandbox with Linux jump/server VMs and support for multiple public IP addresses assigned to Azure Firewall
*
*

### CODE REVIEW

File: 101-azurefirewall-sandbox-linux/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+      }
+    },
+    "location": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]",
+      "metadata": {
+        "description": "Location for all resources."
+      }
+    },
+    "numberOfFirewallPublicIPAddresses": {
```
Comment (bmoore-msft) [inline 101-azurefirewall-sandbox-linux/azuredeploy.json]: is there a min/max here?

File: 101-azurefirewall-sandbox-linux/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+              "nextHopType": "VirtualAppliance",
+              "nextHopIpAddress": "[variables('nextHopIP')]"
+            }
+          }
+        ]
+      }
+    },
+    {
+      "name": "[parameters('virtualNetworkName')]",
+      "apiVersion": "2017-06-01",
```
Comment (bmoore-msft) [inline 101-azurefirewall-sandbox-linux/azuredeploy.json]: update apiVersion

File: 101-azurefirewall-sandbox-linux/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+      "apiVersion": "2017-10-01",
+      "location": "[parameters('location')]",
+      "sku": {
+        "name": "Standard"
+      },
+      "properties": {
+        "publicIPAllocationMethod": "Static",
+        "publicIPAddressVersion": "IPv4"
+      },
+      "copy": {
```
Comment (bmoore-msft) [inline 101-azurefirewall-sandbox-linux/azuredeploy.json]: move this above properties

File: 101-azurefirewall-sandbox-linux/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+            "properties": {
+              "protocol": "Tcp",
+              "sourcePortRange": "*",
+              "destinationPortRange": "22",
+              "sourceAddressPrefix": "*",
+              "destinationAddressPrefix": "*",
+              "access": "Allow",
+              "priority": 1000,
+              "direction": "Inbound",
+              "sourcePortRanges": [],
```
Comment (bmoore-msft) [inline 101-azurefirewall-sandbox-linux/azuredeploy.json]: remove empty properties

File: 101-azurefirewall-sandbox-linux/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,403 @@
...
+              "destinationPortRanges": [],
+              "sourceAddressPrefixes": [],
+              "destinationAddressPrefixes": []
+            }
+          }
+        ]
+      }
+    },
+    {
+      "apiVersion": "2016-03-30",
```
Comment (bmoore-msft) [inline 101-azurefirewall-sandbox-linux/azuredeploy.json]: update apiVersion

File: 101-azurefirewall-sandbox-linux/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,403 @@
...
+            }
+          }
+        ],
+        "networkSecurityGroup": {
+          "id": "[resourceId('Microsoft.Network/networkSecurityGroups', variables('jumpBoxNsgName'))]"
+        }
+      }
+    },
+    {
+      "apiVersion": "2016-03-30",
```
Comment (bmoore-msft) [inline 101-azurefirewall-sandbox-linux/azuredeploy.json]: update apiVersion

File: 101-azurefirewall-sandbox-linux/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,403 @@
...
+          "imageReference": {
+            "publisher": "Canonical",
+            "offer": "UbuntuServer",
+            "sku": "18.04-LTS",
+            "version": "latest"
+          },
+          "osDisk": {
+            "createOption": "FromImage"
+          },
+          "dataDisks": []
```
Comment (bmoore-msft) [inline 101-azurefirewall-sandbox-linux/azuredeploy.json]: remove empty properties

File: 101-azurefirewall-sandbox-linux/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,403 @@
...
+        },
+        "osProfile": {
+          "computerName": "JumpBox",
+          "adminUsername": "[parameters('adminUsername')]",
+          "adminPassword": "[parameters('adminPassword')]",
+          "secrets": []
```
Comment (bmoore-msft) [inline 101-azurefirewall-sandbox-linux/azuredeploy.json]: remove empty properties

File: 101-azurefirewall-sandbox-linux/azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,403 @@
...
+          ]
+        },
+        "diagnosticsProfile": {
+          "bootDiagnostics": {
+            "enabled": true,
+            "storageUri": "[reference(resourceId('Microsoft.Storage/storageAccounts/', variables('storageAccountName'))).primaryEndpoints.blob]"
+          }
+        }
+      },
+      "dependsOn": [
```
Comment (bmoore-msft) [inline 101-azurefirewall-sandbox-linux/azuredeploy.json]: move this above properties

File: 101-azurefirewall-sandbox-linux/azuredeploy.json [hunk 10]
```diff
@@ -0,0 +1,403 @@
...
+          "imageReference": {
+            "publisher": "Canonical",
+            "offer": "UbuntuServer",
+            "sku": "18.04-LTS",
+            "version": "latest"
+          },
+          "osDisk": {
+            "createOption": "FromImage"
+          },
+          "dataDisks": []
```
Comment (bmoore-msft) [inline 101-azurefirewall-sandbox-linux/azuredeploy.json]: remove empty properties

File: 101-azurefirewall-sandbox-linux/azuredeploy.json [hunk 11]
```diff
@@ -0,0 +1,403 @@
...
+        },
+        "osProfile": {
+          "computerName": "Server",
+          "adminUsername": "[parameters('adminUsername')]",
+          "adminPassword": "[parameters('adminPassword')]",
+          "secrets": []
```
Comment (bmoore-msft) [inline 101-azurefirewall-sandbox-linux/azuredeploy.json]: remove empty properties

File: 101-azurefirewall-sandbox-linux/azuredeploy.json [hunk 12]
```diff
@@ -0,0 +1,403 @@
...
+          ]
+        },
+        "diagnosticsProfile": {
+          "bootDiagnostics": {
+            "enabled": true,
+            "storageUri": "[reference(resourceId('Microsoft.Storage/storageAccounts/', variables('storageAccountName'))).primaryEndpoints.blob]"
+          }
+        }
+      },
+      "dependsOn": [
```
Comment (bmoore-msft) [inline 101-azurefirewall-sandbox-linux/azuredeploy.json]: move this above properties

File: 101-azurefirewall-sandbox-linux/azuredeploy.json (hunk 1 above)
Comment (iatodoro) [reply 101-azurefirewall-sandbox-linux/azuredeploy.json]: Added min/max of 1/100 - thanks!

**Comment by iatodoro** [comment]
@bmoore-msft Can this be merged? It's failing some check that I have no visibility to and it works fine on my machine during testing.

**Pull request merged**